File: passenger-service/src/main/java/io/simakkoi9/passengerservice/model/entity/UserStatus.java
package io.simakkoi9.passengerservice.model.entity;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public enum UserStatus {
    ACTIVE(1),
    DELETED(2);

    private final int code;

    public int getCode() {
        return code;
    }

    public static UserStatus fromCode(int code) {
        for (UserStatus status : UserStatus.values()) {
            if (status.getCode() == code) {
                return status;
            }
        }
        throw new IllegalArgumentException();
    }
}

File: driver-service/src/main/kotlin/io/simakkoi9/driverservice/model/entity/UserStatus.kt
package io.simakkoi9.driverservice.model.entity

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

enum class UserStatus(val code: Int) {
    ACTIVE(1),
    DELETED(2);

    companion object {
        @JvmStatic
        @JsonCreator
        fun fromValue(value: String): UserStatus {
            return entries.find {
                it.name.equals(value, ignoreCase = true)
            } ?: throw IllegalArgumentException("Unknown status: $value")
        }

        @JvmStatic
        fun fromCode(code: Int): UserStatus {
            return entries.find {
                it.code == code
            } ?: throw IllegalArgumentException("Unknown code: $code")
        }
    }

    @JsonValue
    fun toValue(): String {
        return this.name
    }
}
